### ISSUE TITLE
is pocketnet working on android?

### ISSUE THREAD

**Issue by GeorvityLabs** [kikirizki/pytorch_to_android-ncnn#1]
hi @kikirizki ,

cool project.
Is pocketnet faster than MobileFaceNet on an android device?

Did you run any tests on an android device.
Looking forward to updates from this project.

**Comment by kikirizki** [comment]
Hi @GeorvityLabs yes there are many backbone for pocketnet it we use small one it is faster than MobileFaceNet, I will update the repo soon, that will include PocketNet and MobileFaceNet, currently I have problem, it seem the facerecognition does not work correctly, maybe I did some mistake

**Comment by GeorvityLabs** [comment]
> Hi @GeorvityLabs yes there are many backbone for pocketnet it we use small one it is faster than MobileFaceNet, I will update the repo soon, that will include PocketNet and MobileFaceNet, currently I have problem, it seem the facerecognition does not work correctly, maybe I did some mistake

Def looking forward to the updated repo.
In case you want to refer MFN projects to fix any issues with your PocketNet android implementation, you can check this article :
https://medium.com/@estebanuri/real-time-face-recognition-with-android-tensorflow-lite-14e9c6cc53a5

**Comment by kikirizki** [comment]
@GeorvityLabs Thank you so much this will surely help

**Comment by kikirizki** [comment]
Hi @GeorvityLabs I have upload the code for MobileFaceNet and BlazeFace (for detection). I tried to add two face to asset folder it seem any random face always has always have high cosine similarity with one face in the asset file. Maybe you could help

**Comment by kikirizki** [comment]
Ah no worries @GeorvityLabs  I have fixed it

**Comment by GeorvityLabs** [comment]
cool work @kikirizki 
Also , I found another pruned mobilefacenet NCNN implementation , I think this will help you as well , in implementing the PocketNet NCNN model.
https://github.com/ywj7373/Face_Recognition_Android

**Comment by GeorvityLabs** [comment]
@kikirizki , do you feel the PocketNet model will be faster than MFN , or do you think current MFN might give higher FPS?

**Comment by GeorvityLabs** [comment]
Also this is the script he has used the pruned the MFN model :
https://github.com/ywj7373/MFN_Pruning

**Comment by kikirizki** [comment]
@GeorvityLabs I believe this pruned MFN will be faster it has very small size

**Comment by kikirizki** [comment]
> cool work @kikirizki Also , I found another pruned mobilefacenet NCNN implementation , I think this will help you as well , in implementing the PocketNet NCNN model. https://github.com/ywj7373/Face_Recognition_Android

I have done implementing pocket net, i will deploy it in next couple hours

**Comment by GeorvityLabs** [comment]
cool , looking forward to updates.
Def curious of which works fastest.

**Comment by GeorvityLabs** [comment]
> > cool work @kikirizki Also , I found another pruned mobilefacenet NCNN implementation , I think this will help you as well , in implementing the PocketNet NCNN model. https://github.com/ywj7373/Face_Recognition_Android
> 
> I have done implementing pocket net, i will deploy it in next couple hours

Is the PocketNet implementation working fine?
Looking forward to have a look at it.

**Comment by kikirizki** [comment]
recently i got wierd error onnx2ncnn on my colab notebook give me `identity not supported yet!` error.Well it did not happened before

**Comment by kikirizki** [comment]
@GeorvityLabs Now pocketnet is working. as you can see on the demo it is faster about 5fps compared to MFN, but it seem less accurate than MFN maybe because I use the smallest pocketnet backbone

**Comment by GeorvityLabs** [comment]
> @GeorvityLabs Now pocketnet is working. as you can see on the demo it is faster about 5fps compared to MFN, but it seem less accurate than MFN maybe because I use the smallest pocketnet backbone

Great work @kikirizki 

Do you have plans to test PocketNetS-256 , PocketNetM-128 & PocketNetM-256.

Technically , they should be faster than MFN right?

Also , is the pruned MFN faster than the current MFN model?

**Comment by GeorvityLabs** [comment]
Also for testing could you add a "+" button like shown in this app : https://github.com/ywj7373/Face_Recognition_Android , so that user can create "user face profile"

**Comment by kikirizki** [comment]
@GeorvityLabs Yes I think 256 will be slightly faster but  i am not sure. Yes I am planning to add 256 too or even  newer face detector and recognizer in very near future. And yes the "+" button also is on my list as next feature

**Comment by GeorvityLabs** [comment]
> @GeorvityLabs Yes I think 256 will be slightly faster but i am not sure. Yes I am planning to add 256 too or even newer face detector and recognizer in very near future. And yes the "+" button also is on my list as next feature

looking forward to the updates.

**Comment by kikirizki** [comment]
> > @GeorvityLabs Yes I think 256 will be slightly faster but i am not sure. Yes I am planning to add 256 too or even newer face detector and recognizer in very near future. And yes the "+" button also is on my list as next feature
> 
> looking forward to the updates.

Thank you for your interest
